joomla/joomla-cms
# Use JHttpFactory as transport mechanism

## Description

This PR is to make use of the available JHttp transport mechanism. So we only have one codebase for doing the downloads instead of two.
## Testing

Please test this PR on a few live hosts as they are more likely to have issues than running this locally.

Test Joomla! Update:
1. Install Joomla 3.4.4 (https://github.com/joomla/joomla-cms/releases/download/3.4.4/Joomla_3.4.4-Stable-Full_Package.zip)
2. Install the patchtester (http://joom.la/patchtester)
3. Apply the patch
4. Go to Components -> Joomla! Updater
5. Check that you get the update 3.4.8 offered
6. Install the 3.4.8 update
7. Check that the update installs fine.

Test Weblinks update:
1. Install the patchtester (http://joom.la/patchtester)
2. Apply the patch
3. Install Weblinks 3.4.0 (https://github.com/joomla-extensions/weblinks/releases/download/3.4.0/pkg_weblinks_3.4.0.zip)
4. Go to Extensions -> Update
5. Scan for updates
6. Install the update Weblinks 3.4.1
7. Check that the update installs fine.

Thanks for @wilsonge for the code review and feedback.
